### ISSUE TITLE
Xiaomi violates the GNU license

### ISSUE THREAD

**Issue by bimizx** [MiCode/Xiaomi_Kernel_OpenSource#2536]
It's understandable that Xiaomi is making money, but time and time again it doesn't release the kernel source code. Copy someone else's work, make a profit, and then give nothing in return. It's been seven months without an update since they last updated the device.It's a monopoly.

The Linux kernel used by Android is GPL-compliant, which means Xiaomi must publish the source code.And the source code released by Xiaomi is incomplete, he has no right to hide the source code.And he only releases the source code once for a mobile phone, disgusting Xiaomi.

Although Xiaomi's infringement does not seem to have much impact on the average Android user, at the development level, it has caused serious damage to the entire ecosystem, as the GNU project illustrates, in this case, the hinder the development of the entire software industry.

Finally, require Xiaomi to abide by the GPL agreement to open source the kernel on time.

**Comment by bimizx** [comment]
Fuck you Xiaomi

**Comment by Molyuu** [comment]
Yes!I stand with you dude

**Comment by Molyuu** [comment]
A lot of issues are started with "please release xxxxx".But guys,remember,Xiaomi SHOULD AND MUST OBEY THE GNU LICENSE AND OPEN SOURCE THEIR KERNEL.

**Comment by sadoyui** [comment]
Fuck you Xiaomi

**Comment by Molyuu** [comment]
So Xiaomi,fuck you!

sadoyui ***@***.***> 于 2022年7月25日周一 11:41写道：

**Comment by xDoge26** [comment]
will device shipped with gki fix this issue ?

**Comment by ashleyirispuppy143** [comment]
hello, ashley from free software foundation here :3

we are going to talk about this to Xiaomi, we still dont know why they dont publish the full source code.

k bye

**Comment by iGOR121** [comment]
> hello, ashley from free software foundation here :3
> 
> we are going to talk about this to Xiaomi, we still dont know why they dont publish the full source code.
> 
> k bye

isn't it obvious that the guy who worked on it just left ximi? no-one's left to do it... just wait for them to get another one

**Comment by Molyuu** [comment]
No.According to the lefted guy who was considered as the head on Weibo(A
Chinese social media platform),he did nothing.

iGOR121 ***@***.***> 于2022年7月30日周六 00:52写道：

**Comment by GEOEGII555** [comment]
I have contacted Xiaomi customer support and they said "we are bonded to the terms and conditions of the company policies". I'm assuming that they meant MediaTek's policy. Tho if it is MediaTek stopping them from publishing the code - does that mean MediaTek is (pretty sure they're doing it knowingly) breaking the GPL?
